obspy/io/nlloc/core.py

```diff
@@ -403,8 +403,8 @@ def write_nlloc_obs(catalog, filename, **kwargs):
         date = pick.time.strftime("%Y%m%d")
         hourminute = pick.time.strftime("%H%M")
         seconds = pick.time.second + pick.time.microsecond * 1e-6
-        time_error = pick.time_errors.uncertainty or -1
-        if time_error == -1:
+        time_error = pick.time_errors.uncertainty or 0.0
+        if time_error == 0.0:
```

Comment:
Is equality to floating `0.0` safe? Should it be `0` or is it the same?

Reply:
Another possibility is:
```python
time_error = pick.time_errors.uncertainty or None
if time_error is None:
  try:
    time_error = (pick.time_errors.upper_uncertainty +
                           pick.time_errors.lower_uncertainty) / 2.0
  except Exception:
    time_error = 0.0
```

Reply:
> Is equality to floating `0.0` safe? Should it be `0` or is it the same?

That's the same.. after all `0.0000000001` won't compare to integer `0`, right? ;-)
Also I don't get the point, is there a floating point implementation that is not able to represent `0` exactly??

Reply:
But yeah, without lookin at the code details your alternative might make sense (I don't know how the upp/lower is generally handled in there)